docs/map.rst

```diff
@@ -0,0 +1,602 @@
+    Map.fromListWith max [("a", 2), ("a", 1), ("b", 2)]
+    > fromList [("a",2),("b",2)]
+
+    -- Build a histogram from a list of elements.
+    Map.fromListWith (+) (map (\x -> (x, 1)) ["a", "a", "b", "c", "c", "c"])
+    > fromList [("a",2),("b",1),("c",3)]
+
+    -- Build a map from a list, combining the string values for the same key.
+    Map.fromListWith (++) [(1, "a"), (1, "b"), (2, "x"), (2, "y")]
+    > fromList [(1,"ba"),(2,"yx")]
+
+
+
+Create a list from a map
+""""""""""""""""""""""""
+
+::
+
+    Map.elems :: Map k v -> [v]
+    Map.elems m = ...
+
+:haddock_short:`Data.Map.Strict#elems` returns a list of values held in the map
+``m``.
+
+::
+
+    Map.toAscList, Map.toList, Map.assocs :: Map k v -> [(k, v)]
+    Map.toAscList m = ...
+
+.. NOTE::
+   These all do the same thing; use ``toAscList`` because its name indicates
+   the ordering.
+
+.. NOTE::
+   ``Map.toList`` is **not** the same as ``Foldable.toList``; the latter is
+   equivalent to ``elems``.
+
+:haddock_short:`Data.Map.Strict#toAscList`,
+:haddock_short:`Data.Map.Strict#toList`, and 
+:haddock_short:`Data.Map.Strict#assocs` returns a list containing the (key,
+value) pairs in the map ``m`` in *ascending* key order.
+
+::
+
+    Map.toDescList :: Map k v -> [(k, v)]
+    Map.toDescList m = ...
+
+:haddock_short:`Data.Map.Strict#toDescList` returns a list containing the (key,
+value) pairs in the map ``m`` in *descending* key order.
+
+::
+
+    Map.elems (Map.fromList [(1,"one"), (2,"two"), (3,"three")])
+    > ["one","two","three"]
+
+    Map.toAscList (Map.fromList [(1,"one"), (2,"two"), (3,"three")])
+    > [(1,"one"),(2,"two"),(3,"three")]
+
+    Map.toDescList (Map.fromList [(1,"one"), (2,"two"), (3,"three")])
+    > [(3,"three"),(2,"two"),(1,"one")]
+
+
+Querying
+^^^^^^^^
+
+Check if a map is empty
+"""""""""""""""""""""""
+
+::
+
+    Map.null :: Map k v -> Bool
+    Map.null m = ...
+
+:haddock_short:`Data.Map.Strict#null` returns ``True`` if the map ``m`` is
+empty, ``False`` otherwise.
+
+::
+
+    Map.null Map.empty
+    > True
+
+    Map.null (Map.fromList [(1,"one")])
+    > False
+
+The number of entries in a map
+""""""""""""""""""""""""""""""
+
+::
+
+    Map.size :: Map k v -> Int
+    Map.size m = ...
+
+:haddock_short:`Data.Map.Strict#size` returns the number of entries in the map
+``m``.
+
+::
+
+    Map.size Map.empty
+    > 0
+
+    Map.size (Map.fromList [(1,"one"), (2,"two"), (3,"three")])
+    > 3
+
+Lookup an entry in the map (lookup)
+"""""""""""""""""""""""""""""""""""
+
+::
+
+    Map.lookup :: Ord k => k -> Map k v -> Maybe v
+    Map.lookup key m = ...
+
+    Map.!? :: Ord k => Map k v -> k -> Maybe v
+    Map.!? m key = ...
+
+:haddock_short:`Data.Map.Strict#lookup` the value corresponding to the given
+``key``, returns ``Nothing`` if the key is not present; the ``!?`` operator
+(*since 0.5.10*) is a flipped version of ``lookup`` and can often be imported
+unqualified.
```

Comment:
`lookup` and `!?` can equally well be imported unqualified (once you hide the silly `lookup` from `Prelude`). I wouldn't necessarily bother mentioning that.

Reply:
Acknowledged.